### PR TITLE
Remove incorrect default value

### LINKED ISSUE #4297: Rolling update is triggered after applying the latest CRD
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

**What did you do?**

Deploy TiDB Operator v1.2.4 and a basic cluster:
```
apiVersion: pingcap.com/v1alpha1
kind: TidbCluster
metadata:
  name: basic
spec:
  version: v5.2.1
  timezone: UTC
  configUpdateStrategy: RollingUpdate
  pd:
    baseImage: pingcap/pd
    replicas: 1
    requests:
      storage: "1Gi"
    config: {}
  tikv:
    baseImage: pingcap/tikv
    replicas: 1
    requests:
      storage: "1Gi"
    config: {}
  tidb:
    baseImage: pingcap/tidb
    replicas: 1
    service:
      type: ClusterIP
    config: {}
```

Upgrade to the latest Operator with the latest CRD
**What did you expect to see?**
Nothing happens to the Pods
**What did you see instead?**
The Pods are rolling updated.

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Close #4297

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 

Upgrade CRD without setting `schedulerName`, check if not rolling upgrade.

- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -218,8 +217,6 @@ type TidbClusterSpec struct {
 	// cluster component is needed to reload the configuration change.
 	// UpdateStrategyRollingUpdate will create a new ConfigMap with the new configuration and rolling-update the
 	// related components to use the new ConfigMap, that is, the new configuration will be applied automatically.
-	// +kubebuilder:validation:Enum=InPlace;RollingUpdate
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go:221]: The validation can be reserved?

Comment (DanielZhangQD) [reply pkg/apis/pingcap/v1alpha1/types.go:221]: The `+kubebuilder:default=IfNotPresent` for ImagePullPolicy should also be removed.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**